### PR TITLE
#593 add password requirements

### LINKED ISSUE #593: Add password requirements
Creating this issue as an extension of https://github.com/rancher/rancher-docs/pull/588.

The password requirements are provider agnostic so a better solution is to list out the password requirements in a central area and link to it rather than repeating them on each provider's page.

Possible pages affected:
- https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/resources/bootstrap-password

Related:
- https://github.com/rancher/rancher-docs/pull/592

### PR DESCRIPTION
**Pull request by martyav**
Fixes #593

## Reminders

- See the [README](../README.md) for more details on how to work with the Rancher docs.

- Verify if changes pertain to other versions of Rancher. If they do, finalize the edits on one version of the page, then apply the edits to the other versions.

- If the pull request is dependent on an upcoming release, make sure to target the release branch instead of `main`.

## Description

This adds the password requirements to a central location and refreshes the **Setting a Bootstrap Password** page to use tabs and remove redundant material.

Also: Users are not actually REQUIRED to reset the admin password after first login, though they likely should.

## Comments

Backticks don't seem to work inside tabs unless they're at the end of a line or on their own line?

A page in the dashboard repo makes some claims about passwords that I'm trying to verify: https://github.com/rancher/dashboard/blob/master/docusaurus/docs/code-base-works/auth-sessions-and-tokens.md

1. ~The minimum password length can be modified by the user (not sure at all if this is relevant to the bootstrap password)~ This doesn't apply to the bootstrap password. Passwords made later must be 12 chars or longer.
2. Password and username must differ

### CODE REVIEW

**Comment by martyav** [comment]
Re: 

> 1. Can users alter the minimum password length post-installation?
> 2. Is it possible for the user to set up Rancher in such a way that the bootstrap password has a minimum length different from 12 chars?
> 3. Are there additional password requirements, such as the username and password must differ?
>
> https://github.com/rancher/dashboard/blob/master/docusaurus/docs/code-base-works/auth-sessions-and-tokens.md from the dashboard repo claims that users can alter the password length with CATTLE_PASSWORD_MIN_LENGTH, but I'm not sure if that field's exposed to users. It also claims the additional requirement that the username and password must differ, which is quite sensible, but wanted to verify.

image

**Comment by martyav** [comment]
@maxsokolovsky Hi, Max. To follow up from this morning, can users alter the minimum length of the admin password (not the special case of the bootstrap -- we have established that it has no min length)? And is there any other validation going on -- for example, requiring that the username and password must differ?

**Comment by maxsokolovsky** [comment]
@martyav, the only thing I am seeing in the code and an actual test is that the password must be 12 characters in length.

**Comment by martyav** [comment]
- https://github.com/rancher/dashboard/issues/6633 - Parent issue

- https://github.com/rancher/dashboard/pull/6801 - v2.6.9
- https://github.com/rancher/dashboard/pull/6897 - v2.7.2:
  - Prevent users to submit password configuration which is not matching the requirement: min 2 and max 256 chars. 
- https://github.com/rancher/dashboard/pull/7930 - v2.7.2: 
  - Added validation for integers exclusively for min-password changes
  - Changed setting key CATTLE_PASSWORD_MIN_LENGTH to PASSWORD_MIN_LENGTH, to match server returned value
  - Allowed editing settings from environment variables; this is due server configuration have been changed for this case
  - Updated tests - https://github.com/rancher/dashboard/pull/7930/commits/3d5a9810e7dfc2e115e38072f0e045ff3d60bf77

File: docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md [hunk 1]
```diff
@@ -6,26 +6,66 @@ title: Setting up the Bootstrap Password
...
+
+When you reset the first admin account's password after first login, the new password must be at least 12 characters long.
+
+## Specifying the Bootstrap Password 
+
+To specify the bootstrap password during initial Rancher installation, run the following commands.
+
+<Tabs>
+<TabItem value="Helm">
+Set the following value in the Rancher Helm chart:
```
Comment (LucasSaintarbor) [inline docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:31]:
```suggestion

Set the following value in the Rancher Helm chart:
```

File: docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md [hunk 2]
```diff
@@ -6,26 +6,66 @@ title: Setting up the Bootstrap Password
...
+
+```yaml
+.Values.bootstrapPassword
 ```
-kubectl get secret --namespace cattle-system bootstrap-secret -o go-template='{{ .data.bootstrapPassword|base64decode}}{{ "\n" }}'
+
+</TabItem>
+<TabItem value="Docker">
+Pass the following value to the Docker install command:
```
Comment (LucasSaintarbor) [inline docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:39]:
```suggestion

Pass the following value to the Docker install command:
```

File: docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md [hunk 3]
```diff
@@ -6,26 +6,66 @@ title: Setting up the Bootstrap Password
...
 
-For a Docker install, you can specify the bootstrap password by passing `-e CATTLE_BOOTSTRAP_PASSWORD=password` to the Docker install command.
+## Retrieving the Bootstrap Password
 
-The password will be stored in the Docker container logs. After Rancher is installed, the UI will show instructions for how to retrieve the password using the Docker container ID:
+The bootstrap password is stored in the Docker container logs. After Rancher is installed, the UI shows instructions for how to retrieve the password based on your installation method. 
 
+<Tabs>
+<TabItem value="Helm">
+To retrieve the password, use kubectl:
```
Comment (LucasSaintarbor) [inline docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:56]:
```suggestion

To retrieve the password, use kubectl:
```

File: docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md [hunk 4]
```diff
@@ -6,26 +6,66 @@ title: Setting up the Bootstrap Password
...
+
+```bash
+kubectl get secret --namespace cattle-system bootstrap-secret -o go-template='{{ .data.bootstrapPassword|base64decode}}{{ "\n" }}'
 ```
+
+</TabItem>
+<TabItem value="Docker">
+To retrieve the password, use the Docker container ID:
```
Comment (LucasSaintarbor) [inline docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:64]:
```suggestion

To retrieve the password, use the Docker container ID:
```

File: docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md [hunk 5]
```diff
@@ -6,26 +6,70 @@ title: Setting up the Bootstrap Password
...
+
+To specify the bootstrap password during initial Rancher installation, run the following commands.
+
+<Tabs>
+<TabItem value="Helm">
+
+Set the following value in the Rancher Helm chart:
+
+```yaml
+.Values.bootstrapPassword
```
Comment (btat) [inline docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:35]: Line 27 says `run the following commands`, but this isn't a command.

Is this intended to be a snippet of the values.yaml file or a snippet of the Helm install command? In either case, the syntax is incorrect.

File: docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md [hunk 6]
```diff
@@ -6,26 +6,70 @@ title: Setting up the Bootstrap Password
...
 ```
-kubectl get secret --namespace cattle-system bootstrap-secret -o go-template='{{ .data.bootstrapPassword|base64decode}}{{ "\n" }}'
+
+</TabItem>
+<TabItem value="Docker">
+
+Pass the following value to the Docker install command:
+
+```bash
+-e CATTLE_BOOTSTRAP_PASSWORD=$PASSWORD
```
Comment (btat) [inline docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:44]: Is `$PASSWORD` intended to be a Bash variable or just a placeholder?

File: docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md [hunk 7]
```diff
@@ -6,26 +6,70 @@ title: Setting up the Bootstrap Password
...
+kubectl get secret --namespace cattle-system bootstrap-secret -o go-template='{{ .data.bootstrapPassword|base64decode}}{{ "\n" }}'
 ```
+
+</TabItem>
+<TabItem value="Docker">
+
+To retrieve the password, use the Docker container ID:
+
+```bash
 docker logs  container-id  2>&1 | grep "Bootstrap Password:"
```
Comment (btat) [inline docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:71]:
```suggestion
docker logs container-id  2>&1 | grep "Bootstrap Password:"
```

**Review by btat — CHANGES_REQUESTED** [review]

File: docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md [hunk 8]
```diff
@@ -6,26 +6,70 @@ title: Setting up the Bootstrap Password
...
+To retrieve the password, use kubectl:
+
+```bash
+kubectl get secret --namespace cattle-system bootstrap-secret -o go-template='{{ .data.bootstrapPassword|base64decode}}{{ "\n" }}'
 ```
+
+</TabItem>
+<TabItem value="Docker">
+
+To retrieve the password, use the Docker container ID:
```
[the remark this answers is not in the document]
Comment (martyav) [reply docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:68]: I'm opting to remove both introductory lines as redundant. I had tried to preserve similar wording as was already on the page, but it isn't really working

File: docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md [hunk 9]
```diff
@@ -6,26 +6,70 @@ title: Setting up the Bootstrap Password
   <link rel="canonical" href="https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/resources/bootstrap-password"/>
 </head>
 
-When Rancher starts for the first time, a password is randomly generated for the first admin user. When the admin first logs in to Rancher, the UI shows commands that can be used to retrieve the bootstrap password. The admin needs to run those commands and log in with the bootstrap password. Then Rancher gives the admin an opportunity to reset the password.
+:::tip
```
[the remark this answers is not in the document]
Comment (martyav) [reply docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md]: Moved and no longer formatted as an info box

**Review by btat — CHANGES_REQUESTED** [review]

File: docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md [hunk 10]
```diff
@@ -6,26 +6,63 @@ title: Setting up the Bootstrap Password
...
+
+## Specifying the Bootstrap Password
+
+<Tabs>
+<TabItem value="Helm">
+
+During [Rancher installation](../install-upgrade-on-a-kubernetes-cluster/install-upgrade-on-a-kubernetes-cluster.md), set `bootstrapPassword` alongside any other flags for the Rancher Helm chart. For example:
+
+```bash
+helm install rancher rancher-<CHART_REPO>/rancher \
```
Comment (btat) [inline docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:33]:
```suggestion
helm install rancher rancher-<chart-repo>/rancher \
```

Placeholder style consistency.

**Review by btat — APPROVED** [review]

**Comment by martyav** [comment]
Synced with v2.8 to v2.6

**Pull request merged**